### PR TITLE
Feature/apvs 222/add bluebird promise library

### PR DESCRIPTION
**Pull request by tmrowe**
Added Bluebird Promise library to the Node applications.
- Added Bluebird Promise library to internal and external web. Not added to external-api-node.
- Promisfied the mongodb dependencies.
- The eligibility-client.js and db-client.js files now return promises.
- The files calling into the eligibility-client.js and db-client.js now expect promises not callbacks.
- Made the external-webs eligibility client take in the name of the collection to perform it's actions against, provided constants in each of the relevant route files, and updated them to call into the refactored functions.
- Updated the collection name variables in internal web to be constants. 
- Removed the logger variable from some files that were not using it.

### CODE REVIEW

File: alpha/internal-web/app/routes/claims.js [hunk 1]
```diff
@@ -2,57 +2,59 @@ var router = require('../routes')
...
+  return {
     reference: 'CL12345TS',
     eligibilityId: eligibilityId,
     approvedEligibilityId: eligibilityId,
     dateSubmitted: moment().format('DD/MM/YYYY'),
     dateApproved: null,
     amount: 45.00,
     claimant: {
-      name: 'Tom Swann'
+      name: 'Tam Swann'
```
Comment (tswann) [inline alpha/internal-web/app/routes/claims.js:46]: 😲

File: alpha/internal-web/app/services/db-client.js [hunk 2]
```diff
@@ -1,67 +1,89 @@
...
-    if (!error) {
-      callback(null, savedRecord)
-    } else {
-      callback(error, null)
-    }
+exports.save = function (claimant, collection) {
+  return new Promise(function (resolve, reject) {
+    mongo.db.collection(collection).insertOne(claimant)
+      .then(function (savedClaimant) {
+        resolve(savedClaimant.ops[0])
```
Comment (tswann) [inline alpha/internal-web/app/services/db-client.js]: Why do we now need the property and indexer? The meaning of ops[0] isn't really obvious here, though I assume it's introduced by Bluebird?

Comment (tmrowe) [reply alpha/internal-web/app/services/db-client.js]: Added a comment to clarify the meaning.

File: alpha/internal-web/app/routes/claims.js (hunk 1 above)
Comment (tmrowe) [reply alpha/internal-web/app/routes/claims.js:46]: Bug fixed! ;)

**Pull request merged**